presto-native-execution/presto_cpp/main/common/Counters.cpp

```diff
@@ -86,6 +86,19 @@ void registerPrestoMetrics() {
       0,
       62l * 1024 * 1024 * 1024, // max bucket value: 62GB
       100);
+
+  /// ================== Cache Counters ==================
+
+  DEFINE_HISTOGRAM_METRIC(
+      kCounterCacheMaxAgeSecs,
```

Comment:
Do we need the histogram here?

Reply:
Histogram gives us MAX, while DEFINE_METRIC("", StatType::AVG) does not support MAX.

For a native worker that we collect data points from over 600 sec, each data point is the max age of all cache entries:
With histogram, we can get the MAX of all max age data points collected over the 600 sec.
With StatType::AVG, we can get the AVG of all max age data points collected over the 600 sec.

So histogram is precise, while StatType::AVG is approximate.

Reply:
Why 600 sec?
Default in Meta is 60 sec.
I've done some testing in the past with AVG and collecting max in the code and exporting it with another AVG.
They were roughly identical.

If you think, most of the counters we use are AVG and they are quite useful.

Imagine you exported 1000 sec at some point and in 30 seconds you exported 1030 sec.
So you would have 1015 sec data point for that minute (60 sec).

The difference is rather irrelevant: 1030 and 1015.

Of course, the can be cliffs:

1000 -> 30 in 30 seconds (when we evict). Then you have ~500, then ~60.
That will also be quite clear.

I don't believe we need second or even minute precision here.
We are talking about data TTL which would be months, weeks or days, unlikely even hours, although hours would work fine here too.

But I don't have a strong opinion here.
We can compare MAX with AVG and see how much delta we've got.